### ISSUE TITLE
Scan Mailbox: Empty "Reply-To:" header leads to javax.mail.internet.AddressException: Illegal address

### ISSUE THREAD

**Issue by sukram230799** [eikek/docspell#606]
Hi,

I ran into the following error while using the `Scan Mailbox` function. I've narrowed it down to one E-Mail and found the cause.

### Analysis

The problem seems to be that if an e-mail address field is present, but does not contain any value, the decoding fails. In this case, it was
```
...
Feedback-Id: customer:cms-customers-purchase_sent_confirmation:Hitmeister
Reply-To:
X-Mailgun-Tag: customers-purchase_sent_confirmation
...
```
After I removed the `Reply-To:` the e-mail went through without any hiccups.

### Error Messages

Log from the joex executor relating to the problem.

``` log
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Next job found: Some(EbwwT7Ne2.../sukram230799/scan-mailbox/Low) 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Creating context for job EbwwT7Ne2.../sukram230799/scan-mailbox/Low to run JobTask(Ident(scan-mailbox),docspell.joex.scheduler.Task$$anonfun$contramap$4@2abc7c97,docspell.joex.scheduler.Task$$anonfun$contramap$4@696466d0) 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Forking job EbwwT7Ne2.../sukram230799/scan-mailbox/Low 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Try to acquire permit (0 free) 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Starting task now 
joex_1        | [ioapp-compute-1] INFO  d.j.s.LogSink - >>> 2021-01-28T14:33:36.903189Z Info EbwwT7Ne2.../sukram230799/scan-mailbox/Low: === Start importing mails for user sukram230799 
joex_1        | [ioapp-compute-5] DEBUG d.j.s.LogSink - >>> 2021-01-28T14:33:36.909500Z Debug EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Settings: {"account":"sukram230799/sukram230799","imapConnection":"mail_provider","folders":["Docspell"],"receivedSince":null,"targetFolder":"INBOX/mail@example.com","deleteMail":false,"direction":null,"itemFolder":null,"fileFilter":"*.pdf|mail.html","tags":null,"subjectFilter":"*Rechnung*|*Beleg*|*Invoice*|*Einzugsverfahren*|*Bestellung*","language":null,"postHandleAll":true} 
joex_1        | [ioapp-compute-1] INFO  d.j.s.LogSink - >>> 2021-01-28T14:33:36.942498Z Info EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Reading mails for user sukram230799 from mail_provider/Docspell 
joex_1        | [ioapp-compute-3] INFO  d.j.s.LogSink - >>> 2021-01-28T14:33:37.802102Z Info EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Processing folder Docspell 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.LogSink - >>> 2021-01-28T14:33:37.937823Z Debug EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Searching next 50 mails in Docspell. 
joex_1        | [ioapp-compute-6] ERROR d.j.s.SchedulerImpl - Job EbwwT7Ne2.../sukram230799/scan-mailbox/Low execution failed. Retrying later. 
joex_1        | javax.mail.internet.AddressException: Illegal address
joex_1        | 	at javax.mail.internet.InternetAddress.<init>(InternetAddress.java:98)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailAddressParse$1(BasicDecode.scala:27)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv.$anonfun$map$1(Conv.scala:10)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$6(BasicDecode.scala:63)
joex_1        | 	at scala.Option.map(Option.scala:242)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$2(BasicDecode.scala:63)
joex_1        | 	at emil.javamail.internal.Util$.$anonfun$withReadFolder$2(Util.scala:26)
joex_1        | 	at emil.javamail.internal.Util$.withOpenFolder(Util.scala:12)
joex_1        | 	at emil.javamail.internal.Util$.withReadFolder(Util.scala:20)
joex_1        | 	at emil.javamail.internal.Util$.$anonfun$withReadFolder$1(Util.scala:26)
joex_1        | 	at scala.Option.map(Option.scala:242)
joex_1        | 	at emil.javamail.internal.Util$.withReadFolder(Util.scala:26)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$1(BasicDecode.scala:52)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.internal.ops.SearchMails$.$anonfun$search$6(SearchMails.scala:65)
joex_1        | 	at scala.collection.immutable.Vector1.map(Vector.scala:1872)
joex_1        | 	at scala.collection.immutable.Vector1.map(Vector.scala:375)
joex_1        | 	at emil.javamail.internal.ops.SearchMails$.$anonfun$search$3(SearchMails.scala:65)
joex_1        | 	at use @ docspell.joex.Main$.run(Main.scala:59)
joex_1        | 	at make @ docspell.common.ThreadFactories$.executorResource(ThreadFactories.scala:42)
joex_1        | 	at make @ docspell.common.ThreadFactories$.executorResource(ThreadFactories.scala:42)
joex_1        | 	at use @ docspell.joex.Main$.run(Main.scala:59)
joex_1        | 	at evalOn @ doobie.util.transactor$Transactor$fromDataSource$FromDataSourceUnapplied.$anonfun$apply$13(transactor.scala:294)
joex_1        | 	at $anonfun$shift$1 @ docspell.joex.scheduler.Task$.docspell$joex$scheduler$Task$$$anonfun$apply$3(Task.scala:50)
joex_1        | 	at $anonfun$shift$1 @ docspell.joex.scheduler.Task$.docspell$joex$scheduler$Task$$$anonfun$apply$3(Task.scala:50)
joex_1        | 	at $anonfun$evalMap_$1 @ fs2.Stream$.$anonfun$resourceWeak$3(Stream.scala:3470)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Job EbwwT7Ne2.../sukram230799/scan-mailbox/Low done Stuck. Releasing resources. 
joex_1        | [ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - New permit acquired 
joex_1        | [ioapp-compute-2] ERROR d.j.s.LogSink - >>> 2021-01-28T14:33:38.355623Z Error EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Job EbwwT7Ne2.../sukram230799/scan-mailbox/Low execution failed. Retrying later. 
joex_1        | javax.mail.internet.AddressException: Illegal address
joex_1        | 	at javax.mail.internet.InternetAddress.<init>(InternetAddress.java:98)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailAddressParse$1(BasicDecode.scala:27)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv.$anonfun$map$1(Conv.scala:10)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$6(BasicDecode.scala:63)
joex_1        | 	at scala.Option.map(Option.scala:242)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$2(BasicDecode.scala:63)
joex_1        | 	at emil.javamail.internal.Util$.$anonfun$withReadFolder$2(Util.scala:26)
joex_1        | 	at emil.javamail.internal.Util$.withOpenFolder(Util.scala:12)
joex_1        | 	at emil.javamail.internal.Util$.withReadFolder(Util.scala:20)
joex_1        | 	at emil.javamail.internal.Util$.$anonfun$withReadFolder$1(Util.scala:26)
joex_1        | 	at scala.Option.map(Option.scala:242)
joex_1        | 	at emil.javamail.internal.Util$.withReadFolder(Util.scala:26)
joex_1        | 	at emil.javamail.conv.BasicDecode.$anonfun$mailHeaderDecode$1(BasicDecode.scala:52)
joex_1        | 	at emil.javamail.conv.Conv$.emil$javamail$conv$Conv$$$anonfun$apply$1(Conv.scala:15)
joex_1        | 	at emil.javamail.conv.Conv$$anonfun$apply$2.convert(Conv.scala:15)
joex_1        | 	at emil.javamail.internal.ops.SearchMails$.$anonfun$search$6(SearchMails.scala:65)
joex_1        | 	at scala.collection.immutable.Vector1.map(Vector.scala:1872)
joex_1        | 	at scala.collection.immutable.Vector1.map(Vector.scala:375)
joex_1        | 	at emil.javamail.internal.ops.SearchMails$.$anonfun$search$3(SearchMails.scala:65)
joex_1        | 	at use @ docspell.joex.Main$.run(Main.scala:59)
joex_1        | 	at make @ docspell.common.ThreadFactories$.executorResource(ThreadFactories.scala:42)
joex_1        | 	at make @ docspell.common.ThreadFactories$.executorResource(ThreadFactories.scala:42)
joex_1        | 	at use @ docspell.joex.Main$.run(Main.scala:59)
joex_1        | 	at evalOn @ doobie.util.transactor$Transactor$fromDataSource$FromDataSourceUnapplied.$anonfun$apply$13(transactor.scala:294)
joex_1        | 	at $anonfun$shift$1 @ docspell.joex.scheduler.Task$.docspell$joex$scheduler$Task$$$anonfun$apply$3(Task.scala:50)
joex_1        | 	at $anonfun$shift$1 @ docspell.joex.scheduler.Task$.docspell$joex$scheduler$Task$$$anonfun$apply$3(Task.scala:50)
joex_1        | 	at $anonfun$evalMap_$1 @ fs2.Stream$.$anonfun$resourceWeak$3(Stream.scala:3470)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | 	at >>$extension @ fs2.concurrent.PubSub$InPartiallyApplied$.$anonfun$from$13(PubSub.scala:220)
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Permit released (0 free) 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Next job found: None 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Waiting for notify 
joex_1        | [ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - Creating context for job EbwwT7Ne2.../sukram230799/scan-mailbox/Low to run cancellation JobTask(Ident(scan-mailbox),docspell.joex.scheduler.Task$$anonfun$contramap$4@2abc7c97,docspell.joex.scheduler.Task$$anonfun$contramap$4@696466d0) 
joex_1        | [ioapp-compute-7] WARN  d.j.s.LogSink - >>> 2021-01-28T14:33:48.382981Z Warn EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Cancelling scan-mailbox task 
joex_1        | [ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - Job EbwwT7Ne2.../sukram230799/scan-mailbox/Low done Cancelled. Releasing resources. 
joex_1        | [ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - Permit released (2 free) 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.SchedulerImpl - Job EbwwT7Ne2.../sukram230799/scan-mailbox/Low has been cancelled. 
joex_1        | [ioapp-compute-0] WARN  d.j.s.LogSink - >>> 2021-01-28T14:33:48.406380Z Warn EbwwT7Ne2.../sukram230799/scan-mailbox/Low: Job has been cancelled. 
joex_1        | [ioapp-compute-7] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/job/EbwwT7Ne2UQ-ekvkQksbKBq-36mdjSL364o-bvmSyp96jax/cancel 
joex_1        | [ioapp-compute-7] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 

```

### Steps to reproduce
1. Create/Manipulate an e-mail with an empty `Header`-field that will be processed as an e-mail address.
2. Put it in the folder that will be scanned
3. Run the `Scan Mailbox`
4. Error occurs

I would fix the error by myself, but I'm not that familiar with the code base.
As this probably also relates to the [emil](https://github.com/eikek/emil) project feel free to reference it :)

**Comment by eikek** [comment]
Thanks a lot for reporting! You are right, this is a bug in the emil library. Basically none of the mail headers fields is required, most usually they are missing completely, but one never knows :)
I'll take a look, I guess there is some `filter(_.nonEmpty)` missing…. If you want some guidance into the code base, let me know.

**Comment by eikek** [comment]
Closed by #642
